web/modules/custom/rdf_entity/src/Entity/Query/Sparql/SparqlCondition.php

```diff
@@ -4,47 +4,785 @@
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\rdf_entity\RdfGraphHandler;
+use Drupal\rdf_entity\RdfFieldHandler;
+use EasyRdf\Serialiser\Ntriples;
 
 /**
  * Defines the condition class for the null entity query.
+ *
+ * @todo: Build a ConditionInterface that extends the ConditionInterface below.
  */
 class SparqlCondition extends ConditionFundamentals implements ConditionInterface {
 
   /**
+   * The rdf graph handler service object.
+   *
+   * @var \Drupal\rdf_entity\RdfGraphHandler
+   */
+  protected $graphHandler;
+
+  /**
+   * The rdf mapping handler service object.
+   *
+   * @var \Drupal\rdf_entity\RdfFieldHandler
+   */
+  protected $fieldHandler;
+
+  /**
+   * Provides a map of filter operators to operator options.
+   *
+   * @var array
+   */
+  protected static $filterOperatorMap = [
+    'IN' => ['delimiter' => ' ', 'prefix' => '', 'suffix' => ''],
+    'NOT IN' => ['delimiter' => ', ', 'prefix' => '', 'suffix' => ''],
+    'IS NULL' => ['use_value' => FALSE],
+    'IS NOT NULL' => ['use_value' => FALSE],
+    'CONTAINS' => ['prefix' => 'FILTER(regex(', 'suffix' => ', "i"))'],
+    'LIKE' => ['prefix' => 'FILTER(regex(', 'suffix' => ', "i"))'],
+    'NOT LIKE' => ['prefix' => 'FILTER(!regex(', 'suffix' => ', "i"))'],
+    'EXISTS' => ['prefix' => 'FILTER EXISTS {', 'suffix' => '}'],
+    'NOT EXISTS' => ['prefix' => 'FILTER NOT EXISTS {', 'suffix' => '}'],
+    // @todo This is not starts with but contains...
+    'STARTS_WITH' => ['prefix' => 'FILTER(regex(', 'suffix' => ', "i"))'],
+    '<' => ['prefix' => '', 'suffix' => ''],
+    '>' => ['prefix' => '', 'suffix' => ''],
+    '>=' => ['prefix' => '', 'suffix' => ''],
+    '<=' => ['prefix' => '', 'suffix' => ''],
+  ];
+
+  /**
+   * The operators that require a default triple to be added.
+   *
+   * In SPARQL, some of the SQL conditions might need more than one conditions
+   * combined. Below are the operators that need a secondary condition.
+   *
+   * @var array
+   *    An array of operators.
+   */
+  protected $requiresTriple = [
+    'IN',
+    'NOT IN',
+    'IS NULL',
+    'IS NOT NULL',
+    'CONTAINS',
+    'LIKE',
+    'NOT LIKE',
+    'STARTS_WITH',
+    '<',
+    '>',
+    '<=',
+    '>=',
+    '!=',
+    '<>',
+  ];
+
+  /**
+   * Whether the conditions have been changed.
+   *
+   * TRUE if the condition has been changed since the last compile.
+   * FALSE if the condition has been compiled and not changed.
+   *
+   * @var bool
+   */
+  protected $changed = TRUE;
+
+  /**
+   * Whether the conditions do not have a triple.
+   *
+   * This will be turned to false if there is at least one condition that does
+   * not involve the id.
+   *
+   * @var bool
+   */
+  protected $needsDefault = TRUE;
+
+  /**
+   * The default bundle predicate.
+   *
+   * @var array
+   */
+  protected $typePredicate = '<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>';
+
+  /**
+   * An array of triples in their string version.
```

Comment:
The 'fragment' suffix was taken as an idea from the core condition. It is not related to the SPARQL. It describes a fragment as a piece of string that will be concatenated in the end of the compilation phase with the rest of the strings to form a string version of the query.